pkg/security/ebpf/c/bind.h

```diff
@@ -0,0 +1,109 @@
+#ifndef _BIND_H_
+#define _BIND_H_
+
+struct bind_event_t {
+    struct kevent_t event;
+    struct process_context_t process;
+    struct span_context_t span;
+    struct container_context_t container;
+    struct syscall_t syscall;
+
+    int socket;
+    uint16_t addr_family;
+    uint16_t addr_port;
+    union  {
+        uint32_t addr_ip;
+        uint8_t  addr_ip6[16];
+    };
+};
+
+SYSCALL_KPROBE3(bind, int, socket, struct sockaddr*, addr, unsigned int, addr_len) {
+    if (!addr) {
+        return 0;
+    }
+
+    struct policy_t policy = fetch_policy(EVENT_BIND);
+    if (is_discarded_by_process(policy.mode, EVENT_BIND)) {
+        return 0;
+    }
+
+    /* cache the bind and wait to grab the retval to send it */
+    struct syscall_cache_t syscall = {
+        .type = EVENT_BIND,
+        .bind = {
+            .socket = socket,
+            .addr = addr,
+            .addr_len = addr_len,
+        },
+    };
+    cache_syscall(&syscall);
+    return 0;
+}
+
+SYSCALL_KRETPROBE(bind) {
```

Comment:
You need to add the tracepoint hook points. These hook points are required to handle kernels that do not support the maxactive parameter. Checkout `tracepoint/syscalls/sys_exit_bpf` for an example with the bpf event.

Reply:
Oh right, thank you! Updated on last commit